src/H5Opline.c

```diff
@@ -110,6 +110,14 @@ H5FL_DEFINE(H5O_pline_t);
  *
  *-------------------------------------------------------------------------
  */
+static char err[] = "ran off the end of the buffer: current p = %p, p_end = %p";
+
+#define VERIFY_LIMIT(p,s,l)                                        \
+  if (p + s - 1 > l) {                                             \
+    HCOMMON_ERROR(H5E_RESOURCE, H5E_NOSPACE, err, p + s, l);       \
```

Comment:
Typically, HCOMMON_ERROR isn't used directly in the code other than being part of the other error macros. Also, most of the error macros should be able to handle printf-like arguments, so these two lines should be able to just be an `HGOTO_ERROR(H5E_RESOURCE, H5E_NOSPACE, "ran off the end of the buffer: current p = %p, p_end = %p", p + s, l)`

Reply:
The use of HCOMMON_ERROR() was a leftover from a previous version of the code - I don't remember why I believed I needed it. Will change this.
The use of the variable` err `instead of the string was to avoid needless repetition of the same string in the binary -  just found out, modern compilers will detect & merge these. 
Will fix and push again.